intents/main.py

```diff
@@ -0,0 +1,186 @@
+        <http://flask.pocoo.org/docs/1.0/api/#flask.Flask.make_response>.
+    """
+    request_json = request.get_json(silent=True)
+    request_args = request.args
+
+    # extracting the intent parameters from the json
+    intent = request_json['intent']
+    table = request_json['table']
+    row_range = request_json['rowRange']
+    metric = request_json['metric']
+    dimensions = request_json['dimensions']
+    summary_operator = request_json['summaryOperator']
+    slices = request_json['slices']
+    is_asc = request_json['isAsc']
+    k = request_json['k']
+    slices = request_json['slices']
+    date = request_json['dateRange']
+    time_granularity = request_json['timeGranularity']
+    row_start = row_range["rowStart"]
+    row_end = row_range["rowEnd"]
+    row_header = row_range["header"]
+
+    # Converting the list of list into a pandas dataframe.
+    query_table = []
+    for row in range(row_start-1, row_end):
+        if row != row_header-1:
+            query_table.append(table[row])
+    query_table_dataframe = pandas.DataFrame(query_table,
+                                             columns=table[row_header-1])
+
+    # Converting the variables that contain denote the
+    # date range into the desired format.
+    date_column_name = None
+    date_range = None
+    if date != "null":
+        date_column_name = date['dateCol']
+        date_range = (date['dateStart'], date['dateEnd'])
+
+    # Converting the Slices passed in the json into a
+    # list of tuples (col, operator, val)
+    slices_list = None
+    if slices != "null":
+        slices_list = []
+        for item in slices:
+            val = item['sliceVal']
+            col = item['sliceCol']
+            operator = _str_to_filter_enum(item['sliceOp'])
+            slices_list.append((col, operator, val))
+
+    if dimensions == 'null':
+        dimensions = None
+
+    if metric == 'null':
+        metric = None
+
+    summary_operator = _str_to_summary_operator_enum(summary_operator)
+
+    suggestions = []
+
+    if intent == 'show':
+        query_table_dataframe = show(query_table_dataframe,
+                                     slices=slices_list,
+                                     metric=metric,
+                                     dimensions=dimensions,
+                                     summary_operator=summary_operator
+                                    )
+    elif intent == 'topk':
+        query_result = topk.topk(query_table_dataframe,
+                                 metric, dimensions, is_asc, k,
+                                 summary_operator=summary_operator,
+                                 date_column_name=date_column_name,
+                                 date_range=date_range,
+                                 slices=slices_list
+                                )
+        query_table_dataframe = query_result[0]
+        suggestions = query_result[1]
+    elif intent == 'slice compare':
```

Comment:
This will be required for other types of intents as well then can remove this for now. Lets add each as we add their supporting functions too.

Reply:
Done